code/modules/projectiles/guns/hookshot.dm

```diff
@@ -506,3 +506,148 @@
 	icon_state = "vampkiller_true"
 	hooktype = /obj/item/projectile/hookshot/whip/vampkiller/true
 
+//Windup-Boxes/////////////////////////////////////////////////////
+/obj/item/weapon/gun/hookshot/whip/windup_box
+	name = "windup-box"
+	icon = 'icons/obj/wind_up.dmi'
+	icon_state = ""
+	item_state = ""
+	desc = ""
+	fire_action = "activate"
+	inhand_states = list()
+	clumsy_check = 0 //Just makes sense
+	force = 5
+	maxlength = 0
+	hooktype = /obj/item/projectile/hookshot/whip/windup_box
+	var/lengthDecider = 0 //replaces maxlength due to a needed reset
+	var/windUp = 0 //amount of times cranked
+	var/maxWindUp = 16 //threshold for overwind
+	var/overWind = 0 //warning/delay system
+	var/state = 0 //Icon changes with each crank, stolen from the crank cell charger
+	var/springForce = 0 //makes the big kicks
+	var/minWindUp = 4
+	var/list/fireSound = null
+	var/fireVolume = 0
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/New()
+	..()
+	maxlength = lengthDecider
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/attack_self(mob/user)
+	if(user.incapacitated())
+		return 1
+	state = !state
+	update_icon()
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/Fire(atom/target, mob/living/user, params, reflex =0, struggle =0) //4 winds minimum
+	maxlength = lengthDecider
+	if(windUp < minWindUp)
+		playsound(src,'sound/items/metal_impact.ogg', 25,1)
+		to_chat(user, "<span class='notice'>the spring isn't tight enough to fire</span>")
+		return
+	playsound(src, fireSound, fireVolume,1)
+	return..()
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/bootbox
+	name = "boot-in-a-box"
+	icon = 'icons/obj/wind_up.dmi'
+	icon_state = "bootbox-0"
+	item_state = "bootbox"
+	desc = "A box with a spring-loaded boot inside. There is a crank attached to wind it up."
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	hooktype = /obj/item/projectile/hookshot/whip/windup_box/bootbox
+	lengthDecider = 4
+	fireSound = 'sound/effects/fence_smash.ogg'
+	fireVolume = 50
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/bootbox/update_icon()
+	icon_state = "bootbox-[state]"
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/bootbox/attack_self(mob/user)
+	if(..())
+		return
```

Comment:
this technically works for now but it would be good if you made it return 1 in case there is ever a child of bootbox (for the same reasons